### DISCUSSION TITLE
Transcribe live streaming audio in AAC

### DISCUSSION THREAD

**Discussion by milkyNik** [deepgram/community#90] [General help] [answered]
Hi

There is a need to send streaming audio in AAC format (2 channels, 44100 sample rate). What header structure is needed? How often should it be sent? I'm interested in the stream scheme: which blocks and in what quantity should be included. I tried sending ADTS + ASC + AAC buffer. Does not work. Need help.

Best regards, Nik

**Comment by BekahHW** [comment]
Hey @milkyNik we'll get back to you on this soon!

**Comment by milkyNik** [comment]
Hi

Want to come back to you regarding header for audio stream and AAC format with more detailed questions. 
Here is the list of open remaining items:
 
Will ADTS header work? https://wiki.multimedia.cx/index.php/ADTS
If ADTS does not work, what is the required header structure?
How frequently the header should be sent?
Is ASC required? http://wiki.multimedia.cx/index.php?title=MPEG-4_Audio#Audio_Specific_Config
If ASC is required, how frequently should it be sent?
Can you send us a header example?
Can you provide an example of schema of sending blocks/buffers? Like: (ADTS Header -> ASC -> AAC Audio buffer) or any other simple expected schema?
Are there any specific parameters expected in the header?

We also tried to send audio in different ways:
as ADTS Header -> ASC -> AAC Audio buffer,
or also as ADTS Header -> AAC Audio buffer,
also as ADTS Header (single time, not looped) -> AAC Audio buffer (looped),
also in this way ASC -> AAC Audio buffer. 

Please let us know if you can assist us.
Thank you,

**Comment by BekahHW** [comment]
Hey @milkyNik,

We would expect it to work if it's containerized, but Deepgram doesn't support AAC format directly for streaming. You’ll need to convert the audio stream to a format like Linear16. The header structure and buffer size will depend on the format of the audio stream, but you can use any header format that's compatible with the Linear16 format. You might try using an open source project like ffmpeg or GStreamer for transcoding.

For the frequency at which the audio stream should be sent, we recommend sending audio in smaller chunks (e.g., one second per chunk) to Deepgram to ensure real-time transcription. The most common chunk size we see in production is 20 ms, so we recommend no smaller than that and no larger than 250 ms. The size of the buffer can vary depending on your use case, but it is recommended to keep the buffer size small to reduce latency.

**Comment by milkyNik** [comment]
Hi @BekahHW 

We are currently streaming audio to Linear16 in chunks of 1024 samples at a sample rate of 44100 (23ms). At the same time, we have a periodic lag of the recognized text from the audio. In rare cases, the lag reaches 10-15 seconds or more. Therefore, we want to use compressed audio streaming to reduce traffic.

What container should we use for streaming AAC audio?

**Comment by milkyNik** [comment]
Hello @BekahHW ,

Just wondering if you could update us on the previous question. 

Regards,

**Reply by BekahHW** [reply]
Have you checked out our docs on [determining audio format](https://developers.deepgram.com/documentation/guides/determine-audio-format/)? I think you'll find that helpful. 

Could you share a little bit more about your project? Why are you trying to stream AAC audio?

**Reply by milkyNik** [reply]
Hello @BekahHW 

We have a video streaming project. Internally, we already use AAC encoding before sending it to streaming services. It is convenient for us to use AAC, especially since Deepgram supports this format https://developers.deepgram.com/documentation/getting-started/audio-formats/

As I wrote above, we want to reduce outgoing traffic in the hope that this will reduce delays in receiving responses from the Deepgram API.

Based on your previous answer, we need to containerize our AAC stream. What is the best way to do this?

If this is not possible for some reason, then tell us which codec we should use and how to containerize it in order to get the best results when working with the Deepgram API

Regards,

**Reply by BekahHW** [reply]
If you want to stream raw audio, [these docs](https://developers.deepgram.com/documentation/features/encoding/) should be helpful. 

Were you able to follow the steps in [the guide](https://developers.deepgram.com/documentation/guides/determine-audio-format/) I linked above? How did that turn out?

If you want to test using AAC, you can try to save a segment of your AAC stream to disk and then send that to Deepgram's batch endpoint. If Deepgram can decode it via the batch endpoint, it should be able to decode it via streaming.

One more thing, circling back to your initial question: When you tried before did you specify the encoding parameter, channels, and sample_rate?

**Comment by jpvajda** [comment, marked as answer]
Closing due to age of issue if this is still a problem just let us know.
